dialogue-core/src/main/java/com/palantir/dialogue/core/BlacklistingChannel.java

```diff
@@ -47,35 +59,119 @@
     @VisibleForTesting
     BlacklistingChannel(LimitedChannel delegate, Duration duration, Ticker ticker) {
         this.delegate = delegate;
-        this.isBlacklisted = Caffeine.newBuilder()
-                .maximumSize(1)
-                .expireAfterWrite(duration)
-                .ticker(ticker)
-                .build();
+        this.duration = duration;
+        this.ticker = ticker;
+        this.perEndpointBlacklistState =
+                Caffeine.newBuilder().maximumSize(1000).ticker(ticker).build();
     }
 
     @Override
     public Optional<ListenableFuture<Response>> maybeExecute(Endpoint endpoint, Request request) {
-        if (isBlacklisted.getIfPresent(KEY) != null) {
-            return Optional.empty();
-        } else {
-            return delegate.maybeExecute(endpoint, request)
-                    .map(future -> DialogueFutures.addDirectCallback(future, new BlacklistingCallback()));
+        BlacklistState state = perEndpointBlacklistState.getIfPresent(endpoint);
+
+        if (state != null) {
```

Comment:
the getIfPresent call on L70 returns null if a channel is not blacklisted, so this check is important to let normal calls go through! (Esp because this isn't a loadingcache, it's just a regular one).

Reply:
Null is neither an instance of BlacklistUntil nor Probation, so it should be fine, right? I may be misreading.

Reply:
ah yep for some reason I thought null would have fallen through those instanceof checks